hazelcast-client/src/main/java/com/hazelcast/client/nearcache/ClientHeapNearCache.java

```diff
@@ -282,10 +271,27 @@ public long getCost() {
                     // object references (key, value, hit)
                     + 3 * (Integer.SIZE / Byte.SIZE);
         }
+
         boolean expired() {
             long time = Clock.currentTimeMillis();
             return (maxIdleMillis > 0 && time > lastAccessTime + maxIdleMillis)
                     || (timeToLiveMillis > 0 && time > creationTime + timeToLiveMillis);
         }
+
+        public int compareTo(CacheRecord<K> o) {
+            if (EvictionPolicy.LRU.equals(evictionPolicy)) {
+                final int result = ((Long) this.lastAccessTime).compareTo((o.lastAccessTime));
```

Comment:
isn't it better to calculate the result with `this.lastAccessTime - o.lastAccessTime` rather than casting to Long ?